src/GitUtils.ps1

```diff
@@ -104,56 +109,54 @@ function Get-GitBranch($gitDir = $(Get-GitDirectory), [Diagnostics.Stopwatch]$sw
-                            'contains' { git describe --contains HEAD 2>$null }
-                            'branch' { git describe --contains --all HEAD 2>$null }
-                            'describe' { git describe HEAD 2>$null }
-                            default { git tag --points-at HEAD 2>$null }
-                        }
-                    } `
-                    {
-                        dbg 'Falling back on parsing HEAD' $sw
-                        $ref = $null
-
-                        if (Test-Path $gitDir\HEAD) {
-                            dbg 'Reading from .git\HEAD' $sw
-                            $ref = Get-Content $gitDir\HEAD 2>$null
-                        }
-                        else {
-                            dbg 'Trying rev-parse' $sw
-                            $ref = git rev-parse HEAD 2>$null
-                        }
-
-                        if ($ref -match 'ref: (?<ref>.+)') {
-                            return $Matches['ref']
-                        }
-                        elseif ($ref -and $ref.Length -ge 7) {
-                            return $ref.Substring(0,7)+'...'
-                        }
-                        else {
-                            return 'unknown'
-                        }
-                    }
-                ) }
+            if ('HEAD' -eq $GitRefName) {
+                $b = '({0})' -f (Invoke-NullCoalescing `
+                  {
+                      dbg 'Trying describe' $sw
+                      switch ($Global:GitPromptSettings.DescribeStyle) {
+                          'contains' { git describe --contains HEAD 2>$null }
+                          'branch' { git describe --contains --all HEAD 2>$null }
+                          'describe' { git describe HEAD 2>$null }
+                          default { git tag --points-at HEAD 2>$null }
+                      }
+                  } `
+                  {
+                      dbg 'Falling back on parsing HEAD' $sw
+                      $ref = $null
+
+                      if (Test-Path $gitDir\HEAD) {
+                          dbg 'Reading from .git\HEAD' $sw
+                          $ref = Get-Content $gitDir\HEAD 2>$null
+                          if ($ref -match 'ref: (?<ref>.+)') {
+                              return $Matches['ref'] -replace 'refs/heads/', ''
+                          }
+                          elseif ($ref -and $ref.Length -ge 7) {
+                              return $ref.Substring(0,7)+'...'
+                          }
+                          else {
+                              return 'unknown'
+                          }
+                      }
+                      else {
+                          return 'unknown'
+                      }
+                  }
+                )
+            }
+            else {
+                $b = $GitRefName
+            }
         }
 
-        dbg 'Inside git directory?' $sw
```

Comment:
I didn't find it (or the other) useful anymore since it was no longer marking a part of the code that takes any time, but I'm happy to put it back if you prefer.